crates/runtime/src/traphandlers.rs

```diff
@@ -167,6 +167,13 @@ cfg_if::cfg_if! {
                 } else if #[cfg(target_os = "macos")] {
                     let cx = &*(cx as *const libc::ucontext_t);
                     (*cx.uc_mcontext).__ss.__rip as *const u8
+                } else if #[cfg(all(target_os = "freebsd", any(target_arch = "x86", target_arch = "x86_64")))] {
+                    let cx = &*(cx as *const libc::ucontext_t);
+                    if #[cfg(target_arch = "x86")] {
```

Comment:
I think the `#[cfg]` expression here is invalid syntax, since that's only valid for the `cfg_if!` macro which doesn't work in a nested fashion like this. Could this be split like 32/64-bit Linux?

Reply:
libc doesn't have ucontext_t for x86, so only x86_64 will work until its ported